### DISCUSSION TITLE
Is there a way to detect the browser through the addon.json?

### DISCUSSION THREAD

**Discussion by Samq64** [ScratchAddons/ScratchAddons#3895] [Help]
There are some options that are not available on Firefox and I was wondering if there was an easy way of detecting the bowser through the addon.json so those option can be greyed out or hidden completely.

**Comment by WorldLanguages** [comment]
Setting `if` property could be extended for this, but first of all, what settings are not available on Firefox?

**Reply by Samq64** [reply]
See #3267 and #3877.

Also, TurboWarp displays "Some addons are not available in this browser: Bitmap images copying" at the bottom of its addon page on Firefox.

**Reply by WorldLanguages** [reply]
> Also, TurboWarp displays "Some addons are not available in this browser: Bitmap images copying" at the bottom of its addon page on Firefox.

That's not a problem in the extension because we can use the navigator.clipboard API on Firefox in extension context.

**Reply by WorldLanguages** [reply]
So it's basically just "dark scrollbars". The Scratch Addons ping can be removed from both browsers.

**Comment by Mbrick2** [comment]
# Opposite 
Also Firefox has a sidebar!!!
